Scripts/RemediationScripts/Remediate-EnableMicrosoftDefenders.ps1

```diff
@@ -0,0 +1,877 @@
+<##########################################
+
+# Overview:
+    This script is used to configure Azure Defender on subscription.
+
+# ControlId: 
+    Azure_Subscription_Config_Enable_MicrosoftDefender_Databases
+    Azure_Subscription_Config_Enable_MicrosoftDefender_ResourceManager
+    Azure_Subscription_Config_Enable_MicrosoftDefender_AppService
+    Azure_Subscription_Config_Enable_MicrosoftDefender_Storage
+    Azure_Subscription_Config_Enable_MicrosoftDefender_Container
+    Azure_Subscription_Config_Enable_MicrosoftDefender_Servers
+    Azure_Subscription_Config_Enable_MicrosoftDefender_KeyVault
+
+# Pre-requisites:
+    1. You will need Owner or Contributor role on subscription.
+    2. Must be connected to Azure with an authenticated account.
+
+# Steps performed by the script:
+    To remediate:
+        1. Install and validate pre-requisites to run the script for subscription.
+        2. Get the list of resource types that do not have Azure Defender plan enabled, from subscription.
+        3. Take a backup of these non-compliant resource types.
+        4. Register 'Microsoft.Security' provider and enable Azure Defender plan for all non-compliant resource types for subscription.
+    
+    To roll back:
+        1. Validate and install the modules required to run the script.
+        2. Get the list of resource type in a Subscription, the changes made to which previously, are to be rolled back.
+        3. Roll back resource type in the Subscription.
+
+# Step to execute script:
+    To remediate:
+        1. Download the script.
+        2. Load the script in a PowerShell session. Refer https://aka.ms/AzTS-docs/RemediationscriptExcSteps to know more about loading the script.
+        3. Execute the script to remediate on AVD Host pool(s) in the Subscription. Refer `Examples`, below.
+
+    To roll back:
+        1. Download the script.
+        2. Load the script in a PowerShell session. Refer https://aka.ms/AzTS-docs/RemediationscriptExcSteps to know more about loading the script.
+        3. Execute the script to rollback on all AVD Host pool(s) in the Subscription. Refer `Examples`, below.
+
+# Command to execute:
+    To remediate:
+        1. Run below command to configure Azure Defender for subscription with all the resource type. 
+           
+            Enable-AzureDefender -SubscriptionId 00000000-xxxx-0000-xxxx-000000000000 -PerformPreReqCheck -EnableAllResourceType
```

Comment:
EnableAllResourceType should be renamed to EnableAllRequiredResourceTypes

Reply:
Done.